safekeeper/src/send_wal.rs

```diff
@@ -498,6 +498,9 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> WalSender<'_, IO> {
             if let Some(stop_pos) = self.stop_pos {
                 if self.start_pos >= stop_pos {
                     // recovery finished
+                    // Note that "ending streaming" part of the string is used by
+                    // pageserver to identify WalReceiverError::SuccessfulCompletion,
+                    // do not change this string without updating pageserver.
```

Comment:
Pageserver should never be the walproposer, which is the postgres compute, see [blog](https://neon.tech/blog/paxos). Or if we sometimes could be, I think we are good to error to omitting it, I would like to learn the situation where this happens.

Reply:
Added a commit that removes this comment.

Reply:
[c1a024a](https://github.com/neondatabase/neon/pull/4902/commits/c1a024a1869195fd3b1dc938ca6bbdd082a3e250) is good, resolving.